### PR TITLE
feat(nimbus): Prevent launching prefFlips experiments to preview

### LINKED ISSUE #10832: prefFlip experiments should not be able to be sent to Preview as the client rejects natual enrollment
**[Prerequisites]:**

* Have access to the Nimbus Console.
* Have a desktop experiment created using the prefFlips feature configuration.
* Set up the latest Nightly to point to the Stage server and the nimbus-preview collection.
  * Set the messaging-system.log pref to all.

**[Steps to reproduce]:**

1. Navigate to the experiment page from prerequisites.
1. Click the Launch to Preview button.
1. Wait for the Experiment to be displayed[ |https://firefox.settings.services.allizom.org/v1/buckets/main/collections/nimbus-preview/records][+here+](https://firefox.settings.services.allizom.org/v1/buckets/main/collections/nimbus-preview/records).
1. Open the Firefox browser from prerequisites.
1. Open the Browser Console and search for the experiment slug.

**[Expected result]:**

* Step 2: An error message is displayed stating that prefFlip experiments can’t be sent to preview.

**[Actual result]:**

* Step 2: The experiment is successfully sent to Preview.

**[Notes]:**

* Attached is a screen recording of the issue.
* Once the experiment is sent to Preview the client enroll in it naturally because it throws an error.
!image-20240612-134950.png|width=892,height=76,alt="image-20240612-134950.png"!
  * While forced enrollment is still possible, bucketing as well as targeting are not testable with this method.
* I’m marking this as an S2 because it blocks a main QA flow and people using the system would think that Preview works just like in other Experiments or Rollouts.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/EXP-4612)
┆Attachments: <a href="https://mozilla-hub.atlassian.net/rest/api/2/attachment/content/64443">image-20240612-134950.png</a>

### PR DESCRIPTION
**Pull request by brennie**
Because:

- Firefox will reject prefFlips experiments from the preview collection

This commit:

- prevents prefFlips experiments from being launched to the preview collection.

Fixes #10832.

### CODE REVIEW

File: experimenter/experimenter/experiments/api/v5/serializers.py [hunk 1]
```diff
@@ -945,6 +945,11 @@ def validate(self, data):
                 }
             )
 
+        if self.should_call_preview_task and not self.instance.can_publish_to_preview:
+            raise serializers.ValidationError(
+                {"status": "This experiment cannot be published to preview."}
```
Comment (jaredlockhart) [inline experimenter/experimenter/experiments/api/v5/serializers.py:950]: Oh I didn't realize we have all these error messages just hanging out in the code here, these should probably be over in constants.  No need to worry about it in this pr tho.

File: experimenter/experimenter/experiments/models.py [hunk 2]
```diff
@@ -822,6 +822,19 @@ def audience_url(self):
...
+        if self.application_config:
+            try:
+                collection = self.kinto_collection
+            except TargetingMultipleKintoCollectionsError:
+                return False
+
+            return (
+                collection is not None
+                and collection == self.application_config.default_kinto_collection
+            )
```
Comment (jaredlockhart) [inline experimenter/experimenter/experiments/models.py:836]: I feel like this coulda just been

```py
            try:
                return self.kinto_collection == self.application_config.default_kinto_collection
            except TargetingMultipleKintoCollectionsError:
                return False
```

?

File: experimenter/experimenter/experiments/tests/factories.py [hunk 3]
```diff
@@ -106,6 +106,14 @@ def schemas(self, create, extracted, **kwargs):
                 )
             )
 
+    @classmethod
+    def create_desktop_prefflips_feature(cls, **kwargs):
```
Comment (jaredlockhart) [inline experimenter/experimenter/experiments/tests/factories.py:110]: Nice 👍

File: experimenter/experimenter/nimbus-ui/src/components/PageSummary/FormLaunchDraftToReview.tsx [hunk 4]
```diff
@@ -31,19 +33,37 @@ const FormLaunchDraftToReview = ({
...
-          <button
-            data-testid="launch-to-preview-instead"
-            type="button"
-            className="btn btn-sm btn-primary"
-            onClick={handleLaunchToPreviewClick}
-          >
-            Launch to Preview Now
-          </button>
-        </p>
+        {canPublishToPreview ? (
```
Comment (jaredlockhart) [inline experimenter/experimenter/nimbus-ui/src/components/PageSummary/FormLaunchDraftToReview.tsx:36]: Yeah this is a good pattern, making a little predicate instance method, and then gating UI on it.  Makes it reusable for when we redo the ui here, makes it easy to change the logic later.  Good pattern.

File: experimenter/experimenter/nimbus-ui/src/hooks/useChangeOperationMutation.tsx [hunk 5]
```diff
@@ -50,6 +50,9 @@ export function useChangeOperationMutation(
                 },
               });
 
+              debugger;
+              console.log(result.data);
```
Comment (jaredlockhart) [inline experimenter/experimenter/nimbus-ui/src/hooks/useChangeOperationMutation.tsx:54]: Sploops 🙈 🙈 🙈

File: experimenter/experimenter/nimbus-ui/src/components/PageSummary/index.tsx [hunk 6]
```diff
@@ -261,13 +261,14 @@ const PageSummary = (props: RouteComponentProps) => {
       >
         {status.draft &&
           !experiment.isArchived &&
-          (showLaunchToReview ? (
+          /* !experiment.canPublishToPreview || */ (showLaunchToReview ? (
```
Comment (jaredlockhart) [inline experimenter/experimenter/nimbus-ui/src/components/PageSummary/index.tsx:264]: Hrmm? What's this here?

**Review by jaredlockhart — APPROVED** [review]
Yep looks good, couple leftover debugging statements, thnx @brennie 🎉 🙏

File: experimenter/experimenter/nimbus-ui/src/components/PageSummary/index.tsx (hunk 6 above)
Comment (brennie) [reply experimenter/experimenter/nimbus-ui/src/components/PageSummary/index.tsx:264]: That was so I could manually test that the correct error message was being returned by the API. Forgot to uncomment this, whoops

**Pull request merged**